.github/workflows/ci.yml

```diff
@@ -12,13 +12,17 @@ jobs:
         with:
           toolchain: 1.56.1
           components: rustfmt, clippy
+      # Pinned to the commit hash of v2.2.1
+      - uses: Swatinem/rust-cache@6fd3edff6979b79f87531400ad694fb7f2c84b1f
+        with:
+          shared-key: test
```

Comment:
In my understanding, [aws-sdk-rust](https://github.com/awslabs/aws-sdk-rust/blob/2ebc092a0372b5f1508b26845c0cf88f537d29b7/.github/workflows/ci.yaml#L45-L48) uses `test` as `shared-key` because it used on `test-sdk` job. However, in our case, the the job is also used for build. Thus, I'd like to propose to use `build-and-test` instead of `test`.

Reply:
Sorry for unresolving this conversation, but IMHO it would be better to squash commit 49eac500f367a5cdc8f1cbd6a497b3a4d6bbc1cb into commit 9e6e632c8091c1b52dfc152f28999a2a09b29823 since they are in a single PR and the "fix" tag is for bug fix in many cases: https://www.conventionalcommits.org/en/v1.0.0/ .

Reply:
I have completed to squash.